wire-ios-data-model/Source/Model/UserClient/E2eIdentityCertificate.swift

```diff
@@ -42,7 +52,9 @@ public enum E2EIdentityCertificateStatus: CaseIterable {
         expiryDate: Date,
         certificateStatus: E2EIdentityCertificateStatus,
         serialNumber: String,
-        comparedDate: Date = DateProvider(now: .now).now
+        comparedDate: Date = DateProvider(now: .now).now,
```

Comment:
This line doesn't really make sense.
Either you inject a date provider (type `DateProviding`) and access the `now` property whenever you need a current date/time, or you simply inject a date:
```suggestion
        comparedDate: Date = .now,
```

Opinion: 
However, in my opinion a certificate shouldn't have any `compareDate` property.
The operations which are defined in the extension below, could be moved to a separate type, like a `E2EIdentityCertificateValidator`, but with a better name.

Reply:
compared date is for aiding unit testing

Reply:
I tried extracting it out. It leads to undesirable changes across the code. May be thing for future if this extension grows